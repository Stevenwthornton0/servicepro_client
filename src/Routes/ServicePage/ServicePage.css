.servicePage {
    width: 85%;
    margin-left: auto;
    margin-right: auto;
    margin-top: 40px;
}

.edit {
    text-decoration: none;
    color: lightblue;
    border: 1px solid black;
    border-radius: 4px;
    padding: 5px 30px;
    background-color: #424242;
}

.edit:hover {
    background-color: lightblue;
    border: 1px solid #424242;
    color: #424242;
    cursor: pointer;
}

.ServicePage_reviews-list {
    margin-top: 40px;
    padding: 0;
}

.ServicePage_reviews-list > li {
    list-style: none;
    border: 1px solid #424242;
    border-radius: 4px;
    margin-bottom: 20px;
    padding-left: 16px;
}

.reviewDelete {
    text-decoration: none;
    color: lightblue;
    border: 1px solid black;
    border-radius: 4px;
    padding: 5px 30px;
    background-color: #424242;
    margin-bottom: 10px;
}

.reviewDelete:hover {
    background-color: lightblue;
    border: 1px solid #424242;
    color: #424242;
    cursor: pointer;
}

.noReviews {
    font-size: 20px;
    font-weight: 500;
}

.line {
    width: 100%;
    border: solid 1.5px #424242;
    margin: 30px auto 30px auto; 
}

@media screen and (min-width: 770px) {
    .servicePage {
        width: 65%;
    }
}
