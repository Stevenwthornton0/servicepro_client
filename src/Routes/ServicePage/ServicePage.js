import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ServiceItem from '../../components/ServiceItem/ServiceItem';
import ServicesService from '../../services/services-api-service';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import ReviewForm from '../../components/ReviewForm/ReviewForm';
import ServicesContext from '../../contexts/ServicesContext';
import './ServicePage.css';

class ServicePage extends Component {
    static contextType = ServicesContext;

    static defaultProps = {
        match: { params: {} }
    }

    componentDidMount() {
        const { serviceId } = this.props.match.params
        ServicesService.getService(serviceId)
            .then(service => {
                this.context.setService(service)
                this.context.setServiceUserId(service.user.id)
            })
        ServicesService.getServiceReviews(serviceId)
            .then(reviews => {
                this.context.setReviews(reviews)
            })
        AuthApiService.getUser(TokenService.getUsername())
            .then(user => {
                this.context.setUser(user.id)
                this.context.setAdmin(user.admin)
            })
    }

    componentWillUnmount() {
        this.context.clearService()
    }

    deleteReview = (reviewId) => {
        const { serviceId } = this.props.match.params;

        ServicesService.deleteReview(serviceId, reviewId)
        ServicesService.getServiceReviews(serviceId)
            .then(reviews => this.context.setReviews(reviews))
    }


    serviceReviews({ reviews, isAdmin=false, deleteReview }) {
        if (reviews[0]) {
            return (
                <ul className='ServicePage_reviews-list'>
                    {reviews.map(review => 
                        <li key={review.id} className='ServicesPage_review'>
                            <p className='bold'>Rating: <span className='notBold'>{review.rating} / 5 Stars</span></p>
                            <p>"{review.text}"</p>
                            <p className='bold'>&#8208; {review.user.first_name} {review.user.last_name}</p>
                            <div className='deleteContainer'>
                                {isAdmin && <button className='reviewDelete' onClick={(e) => deleteReview(review.id)}>Delete</button>}
                            </div>
                        </li>    
                    )}
                </ul>
            )}
            return (
                <p className='noReviews'>No one has posted a review</p>
            )
    }



    ownerEdit = () => {
        const { user, serviceUserId, service } = this.context;

        if (user === serviceUserId) {
            return (
                <Link className='edit' to={`/services/service/${service.id}/edit`}>Edit</Link>
            )
        }

        return null
    }

    render() {
        const { reviews, isAdmin, service } = this.context;

        return (
            <div className='servicePage'>
                <ServiceItem 
                    service={service}
                />

                <this.ownerEdit />

                <this.serviceReviews reviews={reviews} isAdmin={isAdmin} deleteReview={this.deleteReview}/>

                <div className='line'></div>

                <ReviewForm />
            </div>
        )
    }
}

export default ServicePage;