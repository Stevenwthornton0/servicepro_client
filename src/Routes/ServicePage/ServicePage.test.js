import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import ServicePage from './ServicePage';


Enzyme.configure({ adapter: new Adapter() })

describe('ServicePage component', () => {
    it('renders without crashing', () => {
        const wrapper = mount(<ServicePage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

