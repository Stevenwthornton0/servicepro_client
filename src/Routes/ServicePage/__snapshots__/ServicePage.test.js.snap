// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicePage component renders without crashing 1`] = `
<ServicePage
  match={
    Object {
      "params": Object {},
    }
  }
>
  <div>
    <div
      className="loader"
    />
    <div
      className="servicePage"
    >
      <ServiceItem
        service={Array []}
      >
        <div
          className="service"
        >
          <h2 />
          <p
            className="bold"
          >
            Phone: 
          </p>
          <p
            className="bold"
          >
            Email: 
            <span
              className="notBold"
            />
          </p>
          <p />
        </div>
      </ServiceItem>
      <Component />
      <serviceReviews
        deleteReview={[Function]}
        reviews={Array []}
      >
        <p
          className="noReviews"
        >
          No one has posted a review
        </p>
      </serviceReviews>
      <div
        className="line"
      />
      <ReviewForm
        addReview={[Function]}
        service={Array []}
      >
        <div
          className="review"
        >
          <h3>
            Post a Review
          </h3>
          <form
            className="ReviewForm"
            onSubmit={[Function]}
          >
            <label>
              Rating
            </label>
            <select
              className="rating"
              id="rating"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
            <label>
              Review
            </label>
            <textarea
              aria-label="Leave a review..."
              className="reviewText"
              cols="30"
              id="text"
              placeholder="Leave a review.."
              required={true}
              rows="4"
            />
            <button
              className="reviewPost"
              type="submit"
            >
              Post
            </button>
          </form>
        </div>
      </ReviewForm>
    </div>
  </div>
</ServicePage>
`;
